### PR TITLE
style the AddItem component

### LINKED ISSUE #48: Style the "add an item" page
- Design the page according to the [figma file](https://www.figma.com/file/4qFFHuORhrqvQA1bleIRa4/TCL-44?node-id=0%3A1)

### LINKED ISSUE #52: Filter items formatting
1. I don't think we need the filter items label, or it needs to look different if it stays.
filter items

We could change the placeholder text to "Search items..." or "Filter items..."

2. The input size and position doesn´t match the figma design.  The input size of the list page will be different to the others. 
input-size-issue

### PR DESCRIPTION
**Pull request by redapy**
_For an example of how to fill this template out, [see this Pull Request](https://github.com/the-collab-lab/tcl-3-smart-shopping-list/pull/44)._ 

## Description
This PR style the "Add item" page, and remove the "filter text" from the `Search component`
- To make the radio buttons look better, I installed the `custom-forms` plugin.
- I wrapped all the inputs inside their label because I had some issues to style them as they were.
- Styles error and success messages

## Related Issue
Closes #48 
Closes #52 

## Type of Changes

|     | Type                       |
| --- | -------------------------- |
|    | :bug: Bug fix              |
|  ✓ | :sparkles: New feature     |
|    | :hammer: Refactoring       |
|    | :100: Add tests            |
|    ✓| :link: Update dependencies |
|    | :scroll: Docs              |

### Before

before

### After
after

## Testing Steps / QA Criteria
- From your terminal, pull down this branch with `git pull origin rb-style-add-item-page` and check that branch out with `git checkout rb-style-add-item-page`
- run `npm install` to install the new dependecie
- Compare the changes with the Figma file.
## Test searching the list
- In the list page, start your search with a white space.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -12,6 +12,7 @@
     "react-scripts": "^5.0.1"
   },
   "devDependencies": {
+    "@tailwindcss/custom-forms": "^0.2.1",
```
Comment (elitcenk) [inline package.json]: This dependency is not compatible with tailwindcss. Use @tailwindcss/forms instead,  
https://github.com/tailwindlabs/tailwindcss-custom-forms/issues/115

**Review by elitcenk — CHANGES_REQUESTED** [review]

File: package.json (hunk 1 above)
Comment (redapy) [reply package.json]: Thank you @elitcenk , all working now

File: src/components/AddItemForm.js [hunk 2]
```diff
@@ -26,49 +26,61 @@ const AddItemForm = () => {
...
-        <label htmlFor="not-soon">not-Soon</label>
+      </label>
+
+      <fieldset className="flex flex-col gap-y-1 mt-4">
+        <legend className="text-xl">
+          How soon will you buy this item again ?
+        </legend>
+        <label htmlFor="soon">
+          <input
+            className="border-2 border-cyan-500 form-radio text-cyan-500 h-5 w-5 mb-5"
```
Comment (hellodeborahuk) [inline src/components/AddItemForm.js]: Maybe we could use `@apply` here for the repetative classNames for each radio button. I'd also look at the colours so it matches the input field - either both blue or both cyan?

File: src/components/AddItemForm.js [hunk 3]
```diff
@@ -26,49 +26,61 @@ const AddItemForm = () => {
...
-          checked={timeframe === '30'}
-          onChange={handleSelect}
-          name="timeframe"
-        />
-        <label htmlFor="not-soon">not-Soon</label>
+      </label>
+
+      <fieldset className="flex flex-col gap-y-1 mt-4">
+        <legend className="text-xl">
+          How soon will you buy this item again ?
```
Comment (hellodeborahuk) [inline src/components/AddItemForm.js]: Remove space between "again" and "?".

**Review by hellodeborahuk — CHANGES_REQUESTED** [review]
Great work Reda!

File: src/components/AddItemForm.js (hunk 2 above)
Comment (JimeBlue) [reply src/components/AddItemForm.js]: I agree. Using @apply is appropriate here.

Comment (JimeBlue) [reply src/components/AddItemForm.js]: IMO cyan looks better.

**Comment by JimeBlue** [comment]
Excellent job @redapy 🚀

File: src/components/AddItemForm.js (hunk 2 above)
Comment (hellodeborahuk) [reply src/components/AddItemForm.js]: I agree, it goes well with the other colours.

Comment (redapy) [reply src/components/AddItemForm.js]: good idea! I'll use it and also change the input tobe cyan as well

**Comment by redapy** [comment]
what do you think about the colors? I don't if green looks good, was thinking about cyan also for the success message
styles

**Comment by JimeBlue** [comment]
Cyan would be the perfect combination for the success message IMO

**Comment by hellodeborahuk** [comment]
> what do you think about the colors? I don't if green looks good, was thinking about cyan also for the success message 

I agree with Jimena, cyan would look good. I'd just check the colour contrast of the cyan on the white. If it fails, then try a higher number like cyan-600 until it passes the contrast.

**Comment by JimeBlue** [comment]
Great job with the changes @redapy! I love the way it looks with the messages now. This I found very instructive in relation to using aria-label or aria-labelledby [https://tink.uk/the-difference-between-aria-label-and-aria-labelledby/](). Also this [https://www.w3.org/TR/WCAG20-TECHS/ARIA14.html](url)

**Comment by redapy** [comment]
Thanks for the great review. I used `cyan-800` because of a11y, lees than that it will be poor contrast

**Pull request merged**